animation_nodes/id_keys/data_types/integer_type.py

```diff
@@ -169,3 +177,139 @@ def getObjectLocation(self, object):
+     Before :
+     0, 1, 2, 3
+     After :
+     1, 3, 0, 2
+     """)),
+    ('REVERT', "Revert",
+     dedent("""\
+     Revert Index for every selected object which have Index Key.
+     Before :
+     0, 1, 2, 3
+     After :
+     3, 2, 1, 0
+     """)),
+    ('FILL', "Fill Gaps",
+     dedent("""\
+     Fill Gaps in a serie of Index.
+     Before :
+     0, 1, 3, 4
+     After :
+     0, 1, 2, 3
+     """))
+]
+
+randomMethodItems = [
+    ('EXISTING', "Existing",
+     "Randomize existing indexes through selected objects."),
+    ('CREATE', "Create",
+     "Create random index for each selected object.")
+]
+
+class IDKeysIntegerOffset(bpy.types.Operator):
+    bl_idname = "an.id_keys_integer_offset"
+    bl_label = "ID Keys Offset"
+    bl_description = "Offset ID Keys of selected Objects."
+
+    idKeyName: StringProperty()
+    offsetType: EnumProperty(name="Type", default='ADD',
+                            items=offsetTypeItems)
+    offsetValue: IntProperty(name="Offset value", default=0)
+    randomMethod: EnumProperty(name="Method", default='EXISTING',
+                                items=randomMethodItems)
+    randomSeed: IntProperty(name="Seed", default=0)
+    randomMin: IntProperty(name="Min", default=0)
+    randomMax: IntProperty(name="Max", default=10)
+    startAt: BoolProperty(name="Start at", default=False)
+    removeGaps: BoolProperty(name="Remove gaps", default=False)
+
+    @classmethod
+    def poll(cls, context):
+        return len(context.selected_objects) != 0
+
+    def invoke(self, context, event):
+        return context.window_manager.invoke_props_dialog(self, width = 250 * getDpiFactor())
+
+    def draw(self, context):
+        layout = self.layout
+        layout.prop(self, "offsetType", expand = True)
+        if self.offsetType == "ADD":
+            layout.prop(self, "offsetValue")
+        elif self.offsetType == "RANDOMIZE":
+            layout.prop(self, "randomMethod", text = "")
+            if self.randomMethod == "CREATE":
+                row = layout.row(align = True)
+                row.prop(self, "randomMin")
+                row.prop(self, "randomMax")
+            else: layout.prop(self, "randomSeed")
+        elif self.offsetType == "REVERSE":
+            layout.prop(self, "removeGaps")
+            if self.removeGaps:
+                row = layout.row()
+                row.prop(self, "startAt")
+                if self.startAt: row.prop(self, "offsetValue", text = "")
+        elif self.offsetType == "FILL":
+            row = layout.row()
+            row.prop(self, "startAt")
+            if self.startAt: row.prop(self, "offsetValue", text = "")
+
+    def check(self, context):
+        return True
+
+    def execute(self, context):
+        idKey = getIDKeyIntegerName(self.idKeyName)
+        objectList = getObjectsIDKeys(idKey)
+        if self.offsetType == "ADD":
+            for obj in objectList: obj[idKey] = obj[idKey]+self.offsetValue
```

Comment:
Indent the body of the if statement, here and in the following code. Does't look very clear.